docs/rfcs/2023-03-29-file-external-table/rfc.md

```diff
@@ -0,0 +1,86 @@
+---
+Feature Name: "File external table"
+Tracking Issue: https://github.com/GreptimeTeam/greptimedb/issues/1041
+Date: 2023-03-08
+Author: "Xu Wenkang <wenymedia@gmail.com>"
+---
+
+File external table
+
+---
+
+# Summary
+
+Allows users to perform SQL queries on files
+
+# Motivation
+
+User data may already exist in other storages, i.g., file systems/s3, etc. in CSV, parquet, JSON format, etc. We can provide users the ability to perform SQL queries on these files.
+
+
+# Details
+## Overview
+![overview](external-table-engine-overview.png)
+
+We implement a file table engine that creates an external table by accepting user-specified file paths and treating all records as immutable.
+
+1. File Format Decoder: decode files to the `RecordBatch` stream.
+2. Table engine: stores `RecordBatch` in memory and provides scan ability.
```

Comment:
Does the table engine buffers all the content of external table in memory?

Reply:
Yes, If we can't remove these external tables quickly, the buffers might be another problem.

Reply:
> Does the table engine buffers all the content of external table in memory?

> Yes, If we can't remove these external tables quickly, the buffers might be another problem.

I thought about it again, eagerly loading the target format to `RecordBatch` into memory, which might cause unnecessary memory overhead If the user only uses the external target table a few times. Then I change the design to hold the raw file reader and convert the raw file to `RecordBatch` on demand.